### PR TITLE
feat(util-prop-types): add new proptype package

### PR DESCRIPTION
**Pull request by jraff**
## Description

* Publish proptypes as new package @tds/util-prop-types
* Allow componentWithName to work on a prop that contains an array of elements.

### CODE REVIEW

**Comment by jraff** [comment]
```
Changes:
 - @tds/core-notification: 1.1.9 => 1.2.0
 - @tds/util-prop-types: 1.0.0-dev => 1.0.0


🤔 If this is not what you expected, ensure that your commit messages follow the Conventional Commits specification (https://conventionalcommits.org) and try again.

🚀 Please paste the entire output of this task into the body of your Pull Request so that a maintainer can verify before merging/publishing.
```

File: packages/prop-types/prop-types.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable import/prefer-default-export */
+
+export const componentWithName = passedName => {
+  if (typeof passedName !== 'string') {
+    throw new Error('passedName must be a string')
+  }
+  const checkProp = (props, propName, componentName) => {
+    if (Array.isArray(props[propName])) {
+      return props[propName]
+        .map((_, index) => checkProp(props[propName], index, componentName))
```
Comment (theetrain) [inline packages/prop-types/prop-types.js]: Add a comment above this:

// Iterates through every child to see if one of the elements does not match the passed name

File: scripts/e2e.js [hunk 2]
```diff
@@ -18,7 +18,9 @@ const { tdsOptions } = require('./utils/parseArgs')
 const getPackageNames = require('./utils/getPackageNames')
 
 getPackageNames(packageNames => {
-  const onlyCorePackages = packageNames.filter(name => name.startsWith('@tds/core-')).join(' ')
+  const onlyCorePackages = packageNames
+    .filter(name => name.startsWith('@tds/core-') || name === '@tds/util-prop-types')
```
Comment (theetrain) [inline scripts/e2e.js]: This can be reverted

**Comment by jraff** [comment]
```
Changes:
 - @tds/core-notification: 1.1.9 => 1.2.0
 - @tds/util-prop-types: 1.0.0-dev => 1.0.0


🤔 If this is not what you expected, ensure that your commit messages follow the Conventional Commits specification (https://conventionalcommits.org) and try again.

🚀 Please paste the entire output of this task into the body of your Pull Request so that a maintainer can verify before merging/publishing.
```

**Comment by jraff** [comment]
```
Changes:
 - @tds/core-notification: 1.1.9 => 1.2.0
 - @tds/util-prop-types: 1.0.0-dev => 1.0.0


🤔 If this is not what you expected, ensure that your commit messages follow the Conventional Commits specification (https://conventionalcommits.org) and try again.

🚀 Please paste the entire output of this task into the body of your Pull Request so that a maintainer can verify before merging/publishing.
```

**Comment by jraff** [comment]
```
Changes:
 - @tds/core-notification: 1.1.9 => 1.2.0
 - @tds/util-prop-types: 1.0.0-dev => 1.0.0


🤔 If this is not what you expected, ensure that your commit messages follow the Conventional Commits specification (https://conventionalcommits.org) and try again.

🚀 Please paste the entire output of this task into the body of your Pull Request so that a maintainer can verify before merging/publishing.
```

**Pull request merged**